packages/firebase_remote_config/firebase_remote_config/lib/src/firebase_remote_config.dart

```diff
@@ -150,29 +148,16 @@ class FirebaseRemoteConfig extends FirebasePluginPlatform with ChangeNotifier {
       );
     }
   }
-}
-
-@Deprecated('Use FirebaseRemoteConfig instead.')
-class RemoteConfig extends FirebaseRemoteConfig {
-  @Deprecated('Use FirebaseRemoteConfig instead.')
-
-  /// The [FirebaseApp] this instance was initialized with.
-  RemoteConfig._({required FirebaseApp app}) : super._(app: app);
 
-  // Cached instances of [RemoteConfig].
-  static final Map<String, RemoteConfig> _firebaseRemoteConfigInstances = {};
-
-  /// Returns an instance using the default [FirebaseApp].
-  @Deprecated('Use FirebaseRemoteConfig.instance instead.')
-  static RemoteConfig get instance {
-    return RemoteConfig.instanceFor(app: Firebase.app());
-  }
-
-  /// Returns an instance using the specified [FirebaseApp].
-  @Deprecated('Use FirebaseRemoteConfig.instanceFor instead.')
-  static RemoteConfig instanceFor({required FirebaseApp app}) {
-    return _firebaseRemoteConfigInstances.putIfAbsent(app.name, () {
-      return RemoteConfig._(app: app);
-    });
+  /// Starts listening for real-time config updates from the Remote Config backend and automatically
+  /// fetches updates from the RC backend when they are available.
+  ///
+  /// This feature is not supported on Web.
+  ///
+  /// If a connection to the Remote Config backend is not already open, calling this method will
+  /// open it. Multiple listeners can be added by calling this method again, but subsequent calls
+  /// re-use the same connection to the backend.
+  Stream<RemoteConfigUpdate> get onConfigUpdated {
```

Comment:
should we stick to the same naming convention for all firebase plugins? e.g. when it comes to updates, `firebase_auth` has `userChanges()`, `firestore` has `snapshots()` (both are 1. methods, 2. nouns in plural form.

`onConfigUpdated` sounds like an arg that accepts a callback, not a stream

Reply:
I'm not sure there is a naming convention, even inside FlutterFire:
- AppCheck uses `onTokenChange`
- AppInstallation uses `onIdChange`
- Database uses `onChildAdded/Removed/Changed`
- DynamicLinks uses `onLink`
- Messaging uses `onMessage` and `onMessageOpened`
- Storage uses `snapshotEvents` (a getter without on ^^)

I've also look in official packages
- Google Sign In uses [onCurrentUserChanged](https://pub.dev/documentation/google_sign_in/latest/google_sign_in/GoogleSignIn/onCurrentUserChanged.html)
- InAppPurchase uses [purchaseStream](https://pub.dev/documentation/in_app_purchase/latest/in_app_purchase/InAppPurchase/purchaseStream.html)

So even among official packages, there is no real convention. I do find getter more prevalent than methods. Personally, I find getter more obvious since you have to call `listen` on them otherwise, they do nothing whereas a method appears to do something on its own.

Reply:
the fact that official packages don't have a convention doesn't mean we are forbidden to figure out some :) consistency is always a good thing. (GPT nods 😅)

but probably this is out of scope of this PR